server/client_server.go

```diff
@@ -0,0 +1,281 @@
+)
+
+var webhookEvents = []string{"space_created", "space_removed", "space_updated",
+	"page_created", "page_removed", "page_restored", "page_trashed",
+	"page_updated", "comment_created", "comment_removed", "comment_updated"}
+
+type WebhookConfiguration struct {
+	Secret string `json:"secret"`
+}
+
+type CreateWebhookRequestBody struct {
+	Name          string                `json:"name"`
+	Events        []string              `json:"events"`
+	URL           string                `json:"url"`
+	Active        string                `json:"active"`
+	Configuration *WebhookConfiguration `json:"configuration"`
+}
+
+type SpaceResponse struct {
+	ID    int64  `json:"id"`
+	Key   string `json:"key"`
+	Name  string `json:"name"`
+	Links Links  `json:"_links"`
+}
+type CommentContainer struct {
+	ID    string `json:"id"`
+	Type  string `json:"type"`
+	Title string `json:"title"`
+	Links Links  `json:"_links"`
+}
+
+type Links struct {
+	Self string `json:"webui"`
+}
+
+type View struct {
+	Value string `json:"value"`
+}
+type Body struct {
+	View View `json:"view"`
+}
+
+type CreatedBy struct {
+	Username string `json:"username"`
+}
+
+type History struct {
+	CreatedBy CreatedBy `json:"createdBy"`
+}
+
+type CommentResponse struct {
+	ID        string           `json:"id"`
+	Title     string           `json:"title"`
+	Space     SpaceResponse    `json:"space"`
+	Container CommentContainer `json:"container"`
+	Body      Body             `json:"body"`
+	Links     Links            `json:"_links"`
+	History   History          `json:"history"`
+}
+
+type PageResponse struct {
+	ID      string        `json:"id"`
+	Title   string        `json:"title"`
+	Space   SpaceResponse `json:"space"`
+	Body    Body          `json:"body"`
+	Links   Links         `json:"_links"`
+	History History       `json:"history"`
+}
+
+type ConfluenceServerEvent struct {
+	Comment *CommentResponse
+	Page    *PageResponse
+	Space   *SpaceResponse
+	BaseURL string
+}
+
+type confluenceServerClient struct {
+	URL        string
+	HTTPClient *http.Client
+}
+
+type ConfluenceServerUser struct {
+	UserKey     string `json:"userKey"`
+	Username    string `json:"username"`
+	DisplayName string `json:"displayName"`
+}
+
+type AdminData struct {
+	Number int    `json:"number"`
+	Units  string `json:"units"`
+}
+
+type WebhookResponse struct {
+	ID int `json:"id"`
+}
+
+func newServerClient(url string, httpClient *http.Client) Client {
+	return &confluenceServerClient{
+		URL:        url,
+		HTTPClient: httpClient,
+	}
+}
+
+func (csc *confluenceServerClient) CreateWebhook(subscription serializer.Subscription, redirectURL, secret string) (*WebhookResponse, error) {
+	requestBody := &CreateWebhookRequestBody{
+		Name:   subscription.GetAlias(),
+		Events: webhookEvents,
+		URL:    redirectURL,
+		Active: "true",
+		Configuration: &WebhookConfiguration{
+			Secret: secret,
+		},
+	}
+	webhookResponse := &WebhookResponse{}
+	url, err := utils.GetEndpointURL(csc.URL, PathCreateWebhook)
```

Comment:
Couldn't these `GetEnpointURL` and `CallJSON` methods by wrapped into another method with the needed parameters so we don't have to repeat those all the time?

Thank you!